migration/v3/model/file.go

```diff
@@ -0,0 +1,33 @@
+package model
+
+import (
+	"github.com/google/uuid"
+	liberr "github.com/konveyor/controller/pkg/error"
+	"gorm.io/gorm"
+	"os"
+	"path"
+	"time"
+)
+
+type File struct {
+	Model
+	Name       string
+	Path       string `gorm:"<-:create;uniqueIndex"`
+	Expiration *time.Time
+}
+
+func (m *File) BeforeCreate(db *gorm.DB) (err error) {
+	uid := uuid.New()
+	m.Path = path.Join(
+		Settings.Hub.Bucket.Path,
+		".file",
```

Comment:
What's the reason for the dot in the `/.file/` component of the path?

Reply:
It's stored in the bucket root and . prefixed files are ignored by the bucket reaper.  This can be changed when I update the buckets to be more like files.